easybuild/easyblocks/n/namd.py

```diff
@@ -67,6 +57,27 @@ def __init__(self, *args, **kwargs):
         super(EB_NAMD, self).__init__(*args, **kwargs)
         self.namd_arch = None
 
+    def prepare_step(self, *args, **kwargs):
+        """Prepare build environment."""
+        super(EB_NAMD, self).prepare_step(*args, **kwargs)
+
+        if self.cfg['namd_basearch'] is None:
+
+            self.log.info("namd_basearch not specified, so determining it based a CPU arch & CUDA dep...")
+
+            arch = get_cpu_architecture()
+            if arch == X86_64:
+                basearch = 'Linux-x86_64'
+            elif arch == POWER:
+                basearch = 'Linux-POWER'
+
+            cuda = get_software_root('CUDA')
+            if cuda:
+                basearch = '%s.cuda' % basearch
```

Comment:
This breaks the installation of `NAMD-2.14-fosscuda-2019b.eb` that was added by @edmondac in #11090, but I wonder how it ever worked since calling `get_software_root('CUDA')` in `extra_options()` doesn't make sense...

To be revisited.

Reply:
Fails with:

```
ERROR: Build of /tmp/eb-zuwacrgi/files_pr11090/n/NAMD/NAMD-2.14-fosscuda-2019b.eb failed
(err: "build failed (first 300 chars): Failed to patch arch/Linux-x86_64.cuda.tcl:
[Errno 2] No such file or directory: 'arch/Linux-x86_64.cuda.tcl' -> 'arch/Linux-x86_64.cuda.tcl.orig.eb'")
```